frame/support/src/traits/error.rs

```diff
@@ -21,6 +21,12 @@ use sp_std::marker::PhantomData;
 
 /// Trait denoting that the implementing type has the most compact encoded size that is fit to be
 /// included as a field in a variant of the `#[pallet::error]` enum type.
+///
+/// ## Notes
+/// The pallet error enum has a maximum nested depth as defined by
+/// [`frame_support::MAX_NESTED_PALLET_ERROR_DEPTH`]. If the pallet error type exceeds this size
+/// limit, the encoded representation of it will truncate any excess bytes when setting the error
+/// field during the creation of the [`DispatchError`] type.
```

Comment:
maybe there is a way to ensure this at compile-time with some generic `N` on the depth ?

But I'm fine with this. or we can use a vec in dispatch error, a vec would only add one byte per error, it seems ok to me.

The advantage of having a vec would be that the error is directly decodable with type info, no need to do special care when the stuff is truncated.

Reply:
I think we can do something with associated consts, I had been thinking whether I should include an integrity test for the encoded size, but static analysis is always better obviously.